extensions/amp-story/1.0/amp-story-system-layer.css

```diff
@@ -118,13 +121,81 @@
   transform-origin: right !important;
 }
 
+[info] .i-amphtml-message-container {
+  /* 48px for info button, add width to padding for 
+  * additional story buttons */
+  padding-right: 48px !important; 
+}
+
+[info] [dir=rtl] .i-amphtml-message-container {
+  /* 48px for info button, add width to padding for 
+  * additional story buttons */
+  padding-right: auto !important; 
+  padding-left: 48px !important; 
+}
+
+.i-amphtml-message-container {
+  width: max-content !important;
+  transition-property: opacity, transform !important;
+  float: right !important;
+  position: absolute !important;
+  top: 14px !important;
+  right: 48px !important;
+  font-family: 'Roboto', sans-serif !important;
+}
+
+[dir=rtl] .i-amphtml-message-container {
+  right: auto !important;
+  left: 48px !important;
+}
+
+[messagedisplay="noshow"] .i-amphtml-message-container {
+  transition: opacity 0.2s cubic-bezier(0.33, 0.33, 1, 1), transform 0.2s cubic-bezier(0.1, 0.4, 1, 1) 0.5s !important;
+  transform: translateX(10px) !important;
+  opacity: 0 !important;
+}
+
+[dir=rtl][messagedisplay="noshow"] .i-amphtml-message-container {
+  transform: translateX(-10px) !important;
+}
+
+[messagedisplay="show"] .i-amphtml-message-container {
+  transition: opacity 0.2s cubic-bezier(0.33, 0.33, 1, 1), transform 0.2s cubic-bezier(0.33, 0.33, 1, 1) !important;
+  opacity: 1 !important;
+  transform: translateX(0px) !important;
+}
+
+[dir=rtl][messagedisplay="show"] .i-amphtml-message-container {
+  transform: translateX(0px) !important;
+}
+
+.i-amphtml-story-mute-text,
+.i-amphtml-story-unmute-sound-text,
+.i-amphtml-story-unmute-no-sound-text  {
+  width: max-content !important;
+}
+
+.i-amphtml-story-sound-display{
+  display: inline-block !important;
+  height: 46px !important;
+  position: relative !important;
+}
+
+
 .i-amphtml-story-mute-audio-control,
-.i-amphtml-story-unmute-audio-control {
+.i-amphtml-story-unmute-audio-control,
+.i-amphtml-story-mute-text,
+.i-amphtml-story-unmute-sound-text,
+.i-amphtml-story-unmute-no-sound-text  {
   display: none !important;
 }
 
-.audio-playing:not([muted]) .i-amphtml-story-mute-audio-control,
-.audio-playing[muted] .i-amphtml-story-unmute-audio-control {
+.audio-playing[muted='muted'] .i-amphtml-story-unmute-audio-control,
+.audio-playing[muted='muted'] .i-amphtml-story-mute-text,
+.audio-playing[muted='audioon'] .i-amphtml-story-mute-audio-control,
+.audio-playing[muted='audioon'] .i-amphtml-story-unmute-sound-text,
+.audio-playing[muted='noaudio'] .i-amphtml-story-mute-audio-control,
+.audio-playing[muted='noaudio'] .i-amphtml-story-unmute-no-sound-text {
```

Comment:
Can we factorize using `.audio-playing[muted]`?
Not sure if you need to exclude any value for that attribute though.

Reply:
When I try and factorize it with .audio-playing[muted], then I end up seeing both the unmute and mute sound icons, and all 3 of the potential messages when I click on the buttons. The current set up shows only the appropriate icon/message for the correct corresponding audio state. 

Maybe you meant something else and I'm confused so please let me know if this is the case :)